src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,78 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int MAX_CAPACITY = 10;
+
+    private final K[] keys;
+    private final V[] values;
+    private int currentSizeOfStorage;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_CAPACITY];
+        values = (V[]) new Object[MAX_CAPACITY];
+        currentSizeOfStorage = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            for (int i = 0; i < currentSizeOfStorage; i++) {
+                if (keys[i] == null) {
+                    values[i] = value;
+                    return;
+                }
+            }
+        } else {
+            for (int i = 0; i < currentSizeOfStorage; i++) {
+                if (key.equals(keys[i])) {
+                    values[i] = value;
+                    return;
+                }
+            }
```

Comment:
It's better to combine conditions and perform logic in one cycle.
Please use same approach in second method
```suggestion
        for (int i = 0; i < currentSizeOfStorage; i++) {
            if (key == keys[i] || key != null && key.equals(keys[i])) {
                values[i] = value;
                return;
            }
        }
```